## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 477241`
Commit: `348086d3da05a5da80dea32a17aa23809b649ecd`

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.llm-ollama</li>
    <li>python314Packages.llm-ollama.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 74 packages built:</summary>
  <ul>
    <li>python314Packages.langchain-huggingface</li>
    <li>python314Packages.langchain-huggingface.dist</li>
    <li>python314Packages.llm</li>
    <li>python314Packages.llm-anthropic</li>
    <li>python314Packages.llm-anthropic.dist</li>
    <li>python314Packages.llm-cmd</li>
    <li>python314Packages.llm-cmd.dist</li>
    <li>python314Packages.llm-command-r</li>
    <li>python314Packages.llm-command-r.dist</li>
    <li>python314Packages.llm-deepseek</li>
    <li>python314Packages.llm-deepseek.dist</li>
    <li>python314Packages.llm-docs</li>
    <li>python314Packages.llm-docs.dist</li>
    <li>python314Packages.llm-echo</li>
    <li>python314Packages.llm-echo.dist</li>
    <li>python314Packages.llm-fragments-github</li>
    <li>python314Packages.llm-fragments-github.dist</li>
    <li>python314Packages.llm-fragments-pypi</li>
    <li>python314Packages.llm-fragments-pypi.dist</li>
    <li>python314Packages.llm-fragments-reader</li>
    <li>python314Packages.llm-fragments-reader.dist</li>
    <li>python314Packages.llm-fragments-symbex</li>
    <li>python314Packages.llm-fragments-symbex.dist</li>
    <li>python314Packages.llm-gemini</li>
    <li>python314Packages.llm-gemini.dist</li>
    <li>python314Packages.llm-gguf</li>
    <li>python314Packages.llm-gguf.dist</li>
    <li>python314Packages.llm-git</li>
    <li>python314Packages.llm-git.dist</li>
    <li>python314Packages.llm-github-copilot</li>
    <li>python314Packages.llm-github-copilot.dist</li>
    <li>python314Packages.llm-grok</li>
    <li>python314Packages.llm-grok.dist</li>
    <li>python314Packages.llm-groq</li>
    <li>python314Packages.llm-groq.dist</li>
    <li>python314Packages.llm-hacker-news</li>
    <li>python314Packages.llm-hacker-news.dist</li>
    <li>python314Packages.llm-jq</li>
    <li>python314Packages.llm-jq.dist</li>
    <li>python314Packages.llm-llama-server</li>
    <li>python314Packages.llm-llama-server.dist</li>
    <li>python314Packages.llm-mistral</li>
    <li>python314Packages.llm-mistral.dist</li>
    <li>python314Packages.llm-openai-plugin</li>
    <li>python314Packages.llm-openai-plugin.dist</li>
    <li>python314Packages.llm-openrouter</li>
    <li>python314Packages.llm-openrouter.dist</li>
    <li>python314Packages.llm-pdf-to-images</li>
    <li>python314Packages.llm-pdf-to-images.dist</li>
    <li>python314Packages.llm-perplexity</li>
    <li>python314Packages.llm-perplexity.dist</li>
    <li>python314Packages.llm-sentence-transformers</li>
    <li>python314Packages.llm-sentence-transformers.dist</li>
    <li>python314Packages.llm-templates-fabric</li>
    <li>python314Packages.llm-templates-fabric.dist</li>
    <li>python314Packages.llm-templates-github</li>
    <li>python314Packages.llm-templates-github.dist</li>
    <li>python314Packages.llm-tools-datasette</li>
    <li>python314Packages.llm-tools-datasette.dist</li>
    <li>python314Packages.llm-tools-quickjs</li>
    <li>python314Packages.llm-tools-quickjs.dist</li>
    <li>python314Packages.llm-tools-simpleeval</li>
    <li>python314Packages.llm-tools-simpleeval.dist</li>
    <li>python314Packages.llm-tools-sqlite</li>
    <li>python314Packages.llm-tools-sqlite.dist</li>
    <li>python314Packages.llm-venice</li>
    <li>python314Packages.llm-venice.dist</li>
    <li>python314Packages.llm-video-frames</li>
    <li>python314Packages.llm-video-frames.dist</li>
    <li>python314Packages.llm.dist</li>
    <li>python314Packages.manifest-ml</li>
    <li>python314Packages.manifest-ml.dist</li>
    <li>python314Packages.sentence-transformers</li>
    <li>python314Packages.sentence-transformers.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python314Packages.llm-ollama</summary>
<pre>integration_model = &#x27;llama3.2&#x27;

    @pytest.mark.integration
    def test_tools(integration_model):
        &quot;&quot;&quot;Test tool execution.&quot;&quot;&quot;
    
        def multiply(a: int, b: int):
            &quot;Multiply two integers&quot;
            return int(a) * int(b)
    
        model = get_model(integration_model)
        chain = model.chain(&quot;12345 * 4312&quot;, tools=[multiply])
        result = chain.text()
&gt;       assert &quot;53231640&quot; in result or &quot;53,231,640&quot; in result
E       AssertionError: assert (&#x27;53231640&#x27; in &#x27;The result of multiplying 12345 by 4312 is 5,323,16040.&#x27; or &#x27;53,231,640&#x27; in &#x27;The result of multiplying 12345 by 4312 is 5,323,16040.&#x27;)

tests/test_ollama_integration.py:47: AssertionError
=========================== short test summary info ============================
FAILED tests/test_ollama_integration.py::test_tools - AssertionError: assert (&#x27;53231640&#x27; in &#x27;The result of multiplying 12345 by 4...
======================== 1 failed, 48 passed in 20.48s =========================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/accelerate/default.nix
+++ b/pkgs/development/python-modules/accelerate/default.nix
@@ -162,7 +162,15 @@ buildPythonPackage rec {
   ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [
     # RuntimeError: torch_shm_manager: execl failed: Permission denied
     "CheckpointTest"
-  ];
+  ]
+  ++
+    lib.optionals
+      (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64 && (pythonAtLeast "3.14"))
+      [
+        # https://github.com/huggingface/accelerate/issues/3899
+        "test_accelerate_test"
+        "test_cpu"
+      ];
 
   disabledTestPaths = lib.optionals (!(stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64)) [
     # numerous instances of torch.multiprocessing.spawn.ProcessRaisedException:

--- a/pkgs/development/python-modules/llm-ollama/default.nix
+++ b/pkgs/development/python-modules/llm-ollama/default.nix
@@ -22,14 +22,14 @@
 
 buildPythonPackage rec {
   pname = "llm-ollama";
-  version = "0.14.0";
+  version = "0.15.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "taketwo";
     repo = "llm-ollama";
     tag = version;
-    hash = "sha256-8ELjUpHaIC8BOcmyQNeyQDPxQ5vO4Pj6FFnHFhvP+r0=";
+    hash = "sha256-iSLLFC+kYcKq6VOYspkgUyTobU6qUs7FxoAXBmm8H44=";
   };
 
   build-system = [ setuptools ];

--- a/pkgs/development/python-modules/pydevd/default.nix
+++ b/pkgs/development/python-modules/pydevd/default.nix
@@ -10,24 +10,24 @@
   pytest-xdist,
   pytestCheckHook,
   pythonAtLeast,
-  pythonOlder,
   trio,
   untangle,
 }:
 
 buildPythonPackage rec {
   pname = "pydevd";
-  version = "3.3.0";
+  version = "3.4.1";
   pyproject = true;
-
-  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "fabioz";
     repo = "PyDev.Debugger";
-    rev = "pydev_debugger_${lib.replaceStrings [ "." ] [ "_" ] version}";
-    hash = "sha256-V5pM0xnMFnpR1oK0purHFCV3wu+4fOmd72kmy7pVeyk=";
+    tag = "pydev_debugger_${lib.replaceStrings [ "." ] [ "_" ] version}";
+    hash = "sha256-srcYeN4IsnX/B0AWLynr62UC5o+DcjnUrGjcTpvHTCM=";
   };
+
+  # https://github.com/fabioz/PyDev.Debugger/issues/316
+  disabled = pythonAtLeast "3.14";
 
   postPatch = ''
     sed -i '/addopts/d' pytest.ini
@@ -47,6 +47,10 @@ buildPythonPackage rec {
     pytestCheckHook
     trio
     untangle
+  ];
+
+  enabledTestPaths = [
+    "tests/"
   ];
 
   disabledTests = [

